sound/soc/codecs/rt5682.c

```diff
@@ -804,10 +819,27 @@ static const struct snd_kcontrol_new rt5682_if1_45_adc_swap_mux =
 static const struct snd_kcontrol_new rt5682_if1_67_adc_swap_mux =
 	SOC_DAPM_ENUM("IF1 67 ADC Swap Mux", rt5682_if1_67_adc_enum);
 
+static const char * const rt5682_dac_select[] = {
+	"IF1", "SOUND"
+};
+static SOC_ENUM_SINGLE_DECL(rt5682_dacl_enum,
+	RT5682_AD_DA_MIXER, RT5682_DAC1_L_SEL_SFT, rt5682_dac_select);
+
+static const struct snd_kcontrol_new rt5682_dac_l_mux =
+	SOC_DAPM_ENUM("DAC L Mux", rt5682_dacl_enum);
+
+static SOC_ENUM_SINGLE_DECL(rt5682_dacr_enum,
+	RT5682_AD_DA_MIXER, RT5682_DAC1_R_SEL_SFT, rt5682_dac_select);
+
+static const struct snd_kcontrol_new rt5682_dac_r_mux =
+	SOC_DAPM_ENUM("DAC R Mux", rt5682_dacr_enum);
+
 static void rt5682_reset(struct regmap *regmap)
 {
 	regmap_write(regmap, RT5682_RESET, 0);
+#if !IS_ENABLED(CONFIG_SOUNDWIRE)
```

Comment:
It may not work if soundwire is enabled by other sdw codec, but rt5682 works on i2s mode. For example, rt5682 works on i2s mode while rt1308 works on sdw mode. Maybe we can use CONFIG_SND_SOC_RT5682_SDW?